File: MiniCommercialFront/src/app/models/order.ts
import { OrderLine } from "./OrderLine";


export interface Order {
  id?: number;
  orderNumber?: string;
  clientId: number;
  clientName?: string;
  orderDate?: Date;
  status?: string;
  totalHT?: number;
  totalTTC?: number;
  lines: OrderLine[];
}

File: MiniCommercial/MiniCommercial/Models/Order .cs
namespace MiniCommercial.Models.Entities;

public class Order
{
    public int Id { get; set; }
    public string OrderNumber { get; set; } = string.Empty;

    public int ClientId { get; set; } // L'erreur sur .ClientId vient d'ici
    public Client? Client { get; set; }

    public DateTime OrderDate { get; set; } = DateTime.Now;
    public OrderStatus Status { get; set; } = OrderStatus.Brouillon;

    public decimal TotalHT { get; set; } // L'erreur sur .TotalHT
    public decimal TotalTTC { get; set; } // L'erreur sur .TotalTTC

    // L'erreur sur .OrderLines vient d'ici (S bien à la fin)
    public List<OrderLine> OrderLines { get; set; } = new List<OrderLine>();
}